folium/plugins/semicircle.py

```diff
@@ -0,0 +1,108 @@
+
+import json
+
+from branca.element import Figure, JavascriptLink
+
+from folium.map import Marker
+from folium.utilities import _validate_location
+
+from jinja2 import Template
+
+
+class SemiCircle(Marker):
+    """
+    Creates a Semicircle plugin to append into a map with
+    Map.add_plugin.
+    Use (direction and arc) or (start_angle and stop_angle)
+    Parameters
+    ----------
+    location: tuple of length 2, default None
+        The latitude and longitude of the marker.
+        If None, then the middle of the map is used.
+    radius: int, default 0
+        Radius of semicircle in meters
+    direction: int, default 0
+        Heading of direction angle value between 0 and 360 degrees
+    arc: int, default 0
+        Heading of arc angle value between 0 and 360 degrees.
+    start_angle: int, default 0
+        Heading of the start angle value between 0 and 360 degrees
+    stop_angle: int, default 0
+        Heading of the stop angle value between 0 and 360 degrees.
+    fill_color: string, default '#3388ff' (blue)
+        Fill color
+    fill_opacity: float 0-1, default 0.5
+        Fill opacity
+    color: string, default '#3388ff' (blue)
+        Line color
+    opacity: float 0-1, default 1
+        Line opacity
+    **kwargs:
+        Passed to Marker so a popup and tooltip can be provided.
+
+    """
+    _template = Template(u"""
+            {% macro script(this, kwargs) %}
+                if ({{this.direction}} || {{this.arc}}) {
+                    var {{this.get_name()}} = L.semiCircle(
+                        [{{this.location[0]}},{{this.location[1]}}],
+                        {radius:{{this.radius}},
+                        fill: {{this.fill}},
+                        fillColor:'{{this.fill_color}}',
+                        fillOpacity: {{this.fill_opacity}},
+                        color: '{{this.color}}',
+                        opacity: {{this.opacity}}
+                        }).setDirection({{this.direction}},{{this.arc}})
+                        .addTo({{this._parent.get_name()}});
+                } else if ({{this.start_angle}} || {{this.stop_angle}}) {
+                    var {{this.get_name()}} = L.semiCircle(
+                        [{{this.location[0]}},{{this.location[1]}}],
+                        {radius:{{this.radius}},
+                        fill: {{this.fill}},
+                        fillColor:'{{this.fill_color}}',
+                        fillOpacity: {{this.fill_opacity}},
+                        color: '{{this.color}}',
+                        opacity: {{this.opacity}},
+                        startAngle:{{this.start_angle}},
+                        stopAngle:{{this.stop_angle}}
+                        })
+                        .addTo({{this._parent.get_name()}});
+                }
+            {% endmacro %}
+            """)
+
+    def __init__(self, location, radius=0, direction=None, arc=None, start_angle=None, stop_angle=None, fill=True,
```

Comment:
In the `Circle` class that we have there is no default for the `radius` parameter. I think that makes more sense than having a default 0, because that may confuse users; there will be no circle and they won't see why. Can you remove the default 0? That will make radius a required parameter.